### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by jbajou** [spatie/laravel-stats#34] [Q&A]
Hello,

Using this package to compute basic stats about revenue for our customers. I see that the `groupByMonth()` call will hit the database for each month, producing a lot of queries per value.

Is there anyway to optimise this ?
I'm showing 8 metrics, for a little over 100 queries...

**Comment by sneakylenny** [comment]
One way would be caching each stat for a certain time and busting the cache when it updates.
I am too, wondering if it could be optimized.
